### PR TITLE
Fixes for Multi-site docs, fixes #40

### LINKED ISSUE #40: Update multisite recipe for DDEV-Local 1.13.0 (disable-settings-management)
It may be possible to improve the multisite recipe using `--disable-settings-management`. However, there are some cases where it relies on ddev-generated settings. I'm not sure if ddev should maybe have a way to manually trigger generation of CMS settings files while still keeping settings management otherwise-disabled? It would be useful for one-time templating and customization like in this example.

If not, then we can either just leave it alone or expand it to include the necessary settings files.

### PR DESCRIPTION
**Pull request by amitaibu**
Follow up to #42 

I've also started writing a post to explain https://github.com/Gizra/multi-repo , so eventually we'd have an example people could clone, and see in action.

### CODE REVIEW

**Comment by rfay** [comment]
Awesome, thanks!

**Comment by rfay** [comment]
@amitaibu I made some significant changes in new commit https://github.com/drud/ddev-contrib/pull/46/commits/f171b204f89e7e45a9fee0987a89c0011f467843 (and also rebased to current master, make sure to update your fork). 

The most important thing (and what affects everything else) is that I used `--disable-settings-management` so that ddev doesn't try to do any settings at all, so we have to do some things ourselves. 

Looking forward to your review!

**Comment by amitaibu** [comment]
@rfay Thanks. I'm not sure about the use of `--disable-settings-management`.  I was able to avoid it, and benefit from enjoying future DDEV settings, without needing to manually take care of it.

1. See [installation](https://github.com/Gizra/multi-repo#installation) notes. We start with an empty config.yaml, so `ddev composer install` works.
1. Then we create our own `config.local.yaml`, and adapt ddev.settings.php through the [post-start](https://github.com/Gizra/multi-repo/blob/4824ecde5e0256142ef4078ee87664f5fce7ad12/.ddev/config.local.yaml.example#L9-L15) hooks

**Comment by rfay** [comment]
@amitaibu I guess the thing is that ddev's settings management is really only for the base/simple case of everything. And with multisite we've moved way beyond that, so it just gets in the way. And anybody who's doing multisite should be able to do the settings themselves. Also:

* Turning off settings management simplifies things. No removing drush.yml, things like that.
* Some people have complained about ddev creating sites/default in a multisite because they don't use sites/default and don't want it to suddenly appear.

**Comment by amitaibu** [comment]
> No removing drush.yml,

I think that was the only change I had.

> Some people have complained about ddev creating sites/default

That's interesting. I assumed the "right" way to work with multi-sites would be to have a base site, and then for each subsite to spin smaller changes from it. I'm probably too biased towards my use case (and/ or people are doing crazy stuff :wink: ).

But sure, I get your point. I'll do another review later.

File: recipes/drupal8-multisite/README.md [hunk 1]
```diff
@@ -40,80 +45,52 @@ You should get a message that additional domains are now available:
...
+Since the settings.php for each subsite is nearly the same, we can use a basic settings file and include it in each subsite. The example provided here is [web/sites/default/settings.base.php](web/sites/default/settings.base.php). Copy that into the `web/sites/default` directory of your project.
 
 ## Prepare example umami site
 
-Normally, with multisite install, a `settings.php` file is copied from `sites/default` directory. However, DDEV adds its own parallel config file (`settings.ddev.php`). As that ddev file is also auto-managed, just copying it may cause problems later. One of the things it includes is database configuration, so we can't just ignore it. And we also can't just copy `settings.php`, as the inclusion statement is site local and it will not find the additional file back in `sites/default` directory.
-
-For this example, we are just going to include both `settings.php` and `settings.ddev.php` by direct reference and then override some values ([see example](web/sites/umami/settings.php)). The best solution may look a bit different.
+Inspect  [web/sites/umami/settings.php](web/sites/umami/settings.php) and copy it to `web/sites/umami`.
```
Comment (amitaibu) [inline recipes/drupal8-multisite/README.md]:
```suggestion
Inspect [web/sites/umami/settings.php](web/sites/umami/settings.php) and copy it to `web/sites/umami`.
```

File: recipes/drupal8-multisite/README.md [hunk 2]
```diff
@@ -40,80 +45,52 @@ You should get a message that additional domains are now available:
...
 
-1. `cd web/sites/umami`
-2. create `settings.php`
-3. Edit the file to directly include both `sites/default/settings.php` and `sites/default/settings.ddev.php` (if present)
-4. Add override option for default database name, as we gave default "db" user access to all our databases. This keeps things simpler.
-
-Now, if `drush status` is run from within *web* container's `sites/umami` directory, the output should look similar to following:
+Now, if `drush status -l umami.ddev.site` is run from within the *web* container's `sites/umami` directory, the output should look similar to following:
```
Comment (amitaibu) [inline recipes/drupal8-multisite/README.md]:
```suggestion
Now, if `drush status -l umami.ddev.site` is run from within the *web* container's `sites/umami` directory (that is after executing `ddev ssh`), the output should look similar to following:
```

File: recipes/drupal8-multisite/README.md [hunk 3]
```diff
@@ -124,40 +101,40 @@ And reruning `drush status` inside the `sites/umami` directory inside the *web*
 
 ## Install the "basic" demo site
 
-We can follow the same steps for the "basic"" site. We already have the directory, URL, database, and `sites.php` setup. So all that needs to be done is:
+We can follow the same steps as we did for "umami" for the "basic"" site. We already have the directory, URL, database, and `sites.php` setup. So all that needs to be done is:
```
Comment (amitaibu) [inline recipes/drupal8-multisite/README.md]:
```suggestion
We can follow the same steps as we did for "umami" for the "basic" site. We already have the directory, URL, database, and `sites.php` setup. So all that needs to be done is:
```

File: recipes/drupal8-multisite/README.md [hunk 4]
```diff
@@ -124,40 +101,40 @@ And reruning `drush status` inside the `sites/umami` directory inside the *web*
...
 
-1. Copy `settings.php` from `sites/umami` directory and change database name to "basic".
-2. Visit <https://basic.ddev.site> and install the site with the *Standard* profile
+1. Inspect  [web/sites/basic/settings.php](web/sites/basic/settings.php) and copy it to `web/sites/basic`.
```
Comment (amitaibu) [inline recipes/drupal8-multisite/README.md]:
```suggestion
1. Inspect [web/sites/basic/settings.php](web/sites/basic/settings.php) and copy it to `web/sites/basic`.
```

File: recipes/drupal8-multisite/README.md [hunk 5]
```diff
@@ -124,40 +101,40 @@ And reruning `drush status` inside the `sites/umami` directory inside the *web*
...
+2. Visit `https://basic.ddev.site` and install the site with the *Standard* profile
 
 Now, we have a simple Drupal site at <https://basic.ddev.site,> a food magazine at <https://umami.ddev.site> and still only a configuration form at the original <https://d8m.ddev.site.>
 
 ## Set up drush site aliases
 
-[Drush site aliases](http://docs.drush.org/en/9.x/usage/#site-aliases) help to address individual sites without being in the specific directory inside the web container. There is an example *self.site.yml* already with `drush/sites` folder, showing the basic format for default (self) site with environments *prod* and *stage*.
+[Drush site aliases](http://docs.drush.org/en/9.x/usage/#site-aliases) help to address individual sites without being in the specific directory inside the web container.
```
Comment (amitaibu) [inline recipes/drupal8-multisite/README.md]:
```suggestion
[Drush site aliases](http://docs.drush.org/en/9.x/usage/#site-aliases) can address individual sites without being in the specific directory inside the web container.
```

File: recipes/drupal8-multisite/README.md [hunk 6]
```diff
@@ -124,40 +101,40 @@ And reruning `drush status` inside the `sites/umami` directory inside the *web*
...
-3. Copy `settings.php` to the new directory and updating the database name.
+3. Copy `settings.php` to the new directory and update the database name.
 4. Add new alias to the `sites.php`.
 5. Copy a drush alias and updating the `uri` option.
 6. Visit each new site to install Drupal.
 
 ## Conclusion
 
-Of course much more work lies ahead.  The `settings.php` file alone may need a lot more additional configuration. This however, is a standard Drupal issue covered in the documentation.
+Of course much more work lies ahead.  The `settings.php` file alone may need a lot more additional configuration. This however, is a standard Drupal issue covered in Drupal's documentation.
```
Comment (amitaibu) [inline recipes/drupal8-multisite/README.md]: IMO we can remove this sentence, it doesn't help understand, and just gives a feeling you did something wrong.

File: recipes/drupal8-multisite/web/sites/default/settings.base.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+$settings['class_loader_auto_detect'] = FALSE;
+
+// This specifies the default configuration sync directory.
+// $config_directories (pre-Drupal 8.8) and
+// $settings['config_sync_directory'] are supported
+// so it should work on any Drupal 8 or 9 version.
+if (defined('CONFIG_SYNC_DIRECTORY') && empty($config_directories[CONFIG_SYNC_DIRECTORY])) {
+  $config_directories[CONFIG_SYNC_DIRECTORY] = 'sites/default/files/sync';
+} else if (empty($settings['config_sync_directory'])) {
+  $settings['config_sync_directory'] = 'sites/default/files/sync';
```
Comment (amitaibu) [inline recipes/drupal8-multisite/web/sites/default/settings.base.php]:
```suggestion
} 
elseif (empty($settings['config_sync_directory'])) {
  $settings['config_sync_directory'] = 'sites/default/files/sync';
```

**Comment by rfay** [comment]
Yay @amitaibu thanks!

**Pull request merged**